packages/cli/src/main.ts

```diff
@@ -32,7 +33,9 @@ const config: IImperativeConfig = {
     try {
         timingApi.mark("BEFORE_INIT");
 
-        if(process.argv.includes("--daemon")) { config.daemonMode = true; }
+        if(process.argv.includes("--daemon") || (process.env.npm_lifecycle_event != null && process.env.npm_lifecycle_event === "postinstall")) {
```

Comment:
Does `npm_lifecycle_event === "postinstall"`  cover both cases ?
`npm_lifecycle_event != null` and `npm_lifecycle_event === "postinstall"`